### ISSUE TITLE
yarn dev throws long list of errors on typescript  branch

### ISSUE THREAD

**Issue by Bisso1998** [segmentio/evergreen#554]
Hi.
The steps I followed:

1. git clone https://github.com/segmentio/evergreen.git
2. yarn
3. yarn dev

**Everything went fine and I was able to run the storybook on localhost://6006.**
But after this as I did : 
1. git checkout typescript
2. yarn dev
And then a long list of errors were thrown.

`ERROR in ./.storybook/config.js
Module build failed: Error: Cannot find module '@babel/preset-typescript' from '/Users/biswarupbanerjee/Desktop/Dream/evergreen'
    at Function.module.exports [as sync] (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/resolve/lib/sync.js:69:15)
    at resolveStandardizedName (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:101:31)
    at resolvePreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:58:10)
    at loadPreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:77:20)
    at createDescriptor (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at items.map (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPresetDescriptors (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:101:10)
    at passPerPreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:58:96)
    at cachedFunction (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/caching.js:33:19)
    at presets.presets (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:29:84)
    at mergeChainOpts (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:320:26)
    at /Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:283:7
    at buildRootChain (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:68:29)
    at loadPrivatePartialConfig (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/partial.js:85:55)
 @ multi ./node_modules/@storybook/react/dist/server/config/polyfills.js ./node_modules/@storybook/react/dist/server/config/globals.js (webpack)-hot-middleware/client.js?reload=true ./.storybook/config.js

ERROR in ./.storybook/addons.js
Module build failed: Error: Cannot find module '@babel/preset-typescript' from '/Users/biswarupbanerjee/Desktop/Dream/evergreen'
    at Function.module.exports [as sync] (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/resolve/lib/sync.js:69:15)
    at resolveStandardizedName (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:101:31)
    at resolvePreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:58:10)
    at loadPreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/files/plugins.js:77:20)
    at createDescriptor (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at items.map (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPresetDescriptors (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:101:10)
    at passPerPreset (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:58:96)
    at cachedFunction (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/caching.js:33:19)
    at presets.presets (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-descriptors.js:29:84)
    at mergeChainOpts (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:320:26)
    at /Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:283:7
    at buildRootChain (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/config-chain.js:68:29)
    at loadPrivatePartialConfig (/Users/biswarupbanerjee/Desktop/Dream/evergreen/node_modules/@babel/core/lib/config/partial.js:85:55)
 @ multi ./node_modules/@storybook/react/dist/server/config/polyfills.js ./.storybook/addons.js ./node_modules/@storybook/core/dist/client/manager/index.js`

So how to fix this?

**Comment by mshwery** [comment]
Did you install deps on the new branch when you switched? `yarn install` should do the trick.

**Comment by Bisso1998** [comment]
Hi,

Yes I tried doing yarn install in the new branch as well.

But now the error log is as follows:

ERROR in [at-loader] ./src/autocomplete/stories/index.stories.tsx:39:8
    TS2322: Type '{ children: ({ getInputProps, getRef, inputValue }: any) => Element; title: string; onChange: (selectedItem: any) => void; items: any[]; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.
  Property 'title' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.

ERROR in [at-loader] ./src/autocomplete/stories/index.stories.tsx:55:8
    TS2322: Type '{ children: ({ getInputProps, getRef, inputValue, openMenu }: any) => Element; title: string; onChange: (selectedItem: any) => void; items: any[]; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.
  Property 'title' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.

ERROR in [at-loader] ./src/autocomplete/stories/index.stories.tsx:76:8
    TS2322: Type '{ children: ({ getInputProps, getRef, openMenu, inputValue }: any) => Element; onChange: (selectedItem: any) => void; items: any[]; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.
  Property 'onChange' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.

ERROR in [at-loader] ./src/autocomplete/stories/index.stories.tsx:92:8
    TS2322: Type '{ children: ({ getInputProps, openMenu, getRef, inputValue }: any) => Element; isFilterDisabled: true; title: string; onChange: (selectedItem: any) => void; items: any[]; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.
  Property 'title' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Autocomplete> & Pick<Readonly<{ children: (props: any) => ReactNode; }> & Readonly<{ children?: ReactNode; }>, "children"> & Partial<...> & Partial<...>'.

So as we were discussing in the spectrum chat we can either ignore typecheck in the stories or find/fix the type broken types.

**Comment by mshwery** [comment]
Yeah these are two separate types of issues @Bisso1998 

The first errors (in the issue) was due to not having npm modules installed (via `yarn install`) on the new branch yet.

The other errors are TypeScript errors in the stories. We started adding real types to some of the components, but haven't fully converted the stories to match apparently. 

This is a great opportunity to fix the stories to use the types properly, and update the types where they need to be adjusted.

Since this isn't a bug with any evergreen releases or the dev commands I'm going to close this issue. Feel free to introduce PRs to resolve some of the mis-used types in the stories!
